docs/src/pages/components/dialogs/DialogWithPortalsInside.js

```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { Button, Dialog, TextField, Popper } from "@material-ui/core";
+
+function App() {
+  const [dialogOpen, setDialogOpen] = React.useState(false);
+  const [popperOpen, setPopperOpen] = React.useState(false);
+  const [buttonRef, setButtonRef] = React.useState(null);
+
+  const [secondPopperOpen, setSecondPopperOpen] = React.useState(false);
+  const [secondButtonRef, setSecondButtonRef] = React.useState(null);
+
+  return (
+    <div className="App">
+      <Button onClick={() => setDialogOpen(true)}>Open dialog</Button>
+      <Dialog open={dialogOpen} onClose={() => setDialogOpen(false)}>
+        <Button
+          ref={ref => setButtonRef(ref)}
+          onClick={() => setPopperOpen(!popperOpen)}
+        >
+          Open popper
+        </Button>
+
+        <Popper open={popperOpen} anchorEl={buttonRef} style={{zIndex: 2000}}>
+          <Button
+            ref={ref => setSecondButtonRef(ref)}
+            onClick={() => { setSecondPopperOpen(!secondPopperOpen)}}
+            onMouseUp={() => {console.log("Second button clicked")}}
+          >
+            Open popper
+          </Button>
+          <TextField
+            autoFocus
+            onFocus={() => console.log("TextField focus")}
+            onBlur={() => console.log("TextField blur")}
+          />
+          <Popper open={secondPopperOpen} anchorEl={secondButtonRef} style={{zIndex: 2001}}>
```

Comment:
`zIndex` is required here, because it needs to be bigger than the one of the backdrop, as click events are not propagated to the correct element otherwise.